File: app/api/v1/student.py
import uuid
from typing import Annotated

from fastapi.responses import JSONResponse

from app.api.dependencies import require_student, require_teacher
from app.common.response import response_success
from app.schema.user import TokenData
from app.service.student_service import StudentService
from fastapi import APIRouter, Depends

router = APIRouter()


@router.get("/weekly-study", description="学生获取本周每日学习时长(秒)")
async def get_student_weekly_study(
    payload: Annotated[TokenData, Depends(require_student)],
    student_svc: StudentService = Depends(StudentService),
) -> JSONResponse:
    seconds = await student_svc.get_weekly_study(payload.id)
    return response_success({"daily_seconds": seconds})


@router.get(
    "/{student_id}/course/{course_id}/weekly-study", description="教师获取指定学生在某课程的本周每日学习时长(秒)"
)
async def get_student_weekly_study_by_teacher(
    student_id: str,
    course_id: uuid.UUID,
    payload: Annotated[TokenData, Depends(require_teacher)],
    student_svc: StudentService = Depends(StudentService),
) -> JSONResponse:
    seconds = await student_svc.get_weekly_study_in_course(student_id, course_id, payload.id)
    return response_success({"daily_seconds": seconds})


@router.get("/tasks-todo", description="学生获取待办学习任务列表")
async def get_student_tasks_todo(
    payload: Annotated[TokenData, Depends(require_student)],
    student_svc: StudentService = Depends(StudentService),
) -> JSONResponse:
    data = await student_svc.get_tasks_todo(payload.id)
    return response_success(data)

File: source_code_front/src/api/student.js
import request from "@/utils/request.js";

/**
 * @description 获取学生本周每日学习时长 #给学生端使用，直接获取一周每天的学习时长数据
 * @method GET /student/weekly-study
 * @response { code: number, data: number[] }
 * @interactionNotes data 直接是一个长度为 7 的浮点数数组（代表周一到周日的小时数，如 [1.5, 2.0, 0.5, ...]），不再有 daily_hours 嵌套。
 */
export function get_student_weekly_study() {
    return request.get(`/student/weekly-study`);
}

/**
 * 获取某学生在某课程中的周学习时长（周一至周日，共7天）
 * @param {number|string} courseId - 课程ID
 * @param {number|string} studentId - 学生ID
 * @returns {Promise} 信封结构响应
 */
export const get_student_weekly_study_in_course = (courseId, studentId) => {
    return request({
        url: `/student/${studentId}/course/${courseId}/weekly-study`,
        method: 'get'
    })
}